File: backend/src/main/java/com/lifeos/observability/log/dto/TrackerLogDto.java
package com.lifeos.observability.log.dto;

import lombok.*;

import java.time.Instant;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TrackerLogDto {

    private Instant timestamp;

    private String level;

    private String tag;

    private String eventType;

    private String message;
}

File: android/app/src/main/java/com/karthik/lifeos/tracker/sync/dto/TrackerLogDto.kt
package com.karthik.lifeos.tracker.sync.dto

import java.time.Instant

/**
 * DTO matching the backend's TrackerLogDto.
 */
data class TrackerLogDto(
    val timestamp: String,     // ISO-8601 Instant string
    val level: String,
    val tag: String,
    val eventType: String?,
    val message: String
) {
    companion object {
        fun fromEpochMillis(millis: Long): String {
            return Instant.ofEpochMilli(millis).toString()
        }
    }
}
